socket.go

```diff
@@ -273,6 +275,12 @@ func (socket *mongoSocket) Release() {
 				recycle = false
 			}
 		}
+		if socket.expiryTime != nil && time.Now().After(*socket.expiryTime) {
+			stats.socketsExpired(+1)
```

Comment:
Its a convention that I'm following that is used throughout the code e.g. https://github.com/lyft/mgo/blob/v2-unstable/socket.go#L191